### PR TITLE
adpd410x_iio: add read samples support

### PR DESCRIPTION
**Pull request by amiclaus**
Add read samples function required by the IIO buffer support of the
device.

Signed-off-by: Antoniu Miclaus <antoniu.miclaus@analog.com>

### CODE REVIEW

**Review by dbogdan — APPROVED** [review]

File: drivers/photo-electronic/adpd410x/iio_adpd410x.c [hunk 1]
```diff
@@ -84,6 +84,41 @@ static int adpd410x_iio_read_raw_chan(void *device, char *buf, uint32_t len,
...
+ * @param nb_samples - Input number of samples.
+ * @return Number of samples, or negative code.
+ */
+static int32_t adpd410x_read_samples(void *device, uint32_t *buff,
+				     uint32_t nb_samples)
+{
+	struct adpd410x_dev *dev = (struct adpd410x_dev *)device;
+
+	int32_t ret;
+	uint32_t data[8], i, offset = 0;
```
Comment (adrimbarean) [inline drivers/photo-electronic/adpd410x/iio_adpd410x.c:100]: nitpick: could we use a constant for 8 here and use it in all other places? Will make it easier to maintain in the future.

**Review by adrimbarean — APPROVED** [review]
Looks good to me. Just a comment.

File: drivers/photo-electronic/adpd410x/iio_adpd410x.c (hunk 1 above)
Comment (amiclaus) [reply drivers/photo-electronic/adpd410x/iio_adpd410x.c:100]: sure, I can add a commit prior to this one, with the suggested change.

**Review by amiclaus — COMMENTED** [review]
v2:
- add commit for number of IIO channels definition

**Pull request merged**